// File compilation command: python3 -m grpc_tools.protoc -I./proto --python_out=. --grpc_python_out=. ./proto/judge.proto

syntax = "proto3";

service JudgeService {
    rpc judge (SubmissionRequest) returns (SubmissionResult) {}
}

message SubmissionRequest {
    string username = 1;
    string source = 2;
    string lang = 3;
    string problem = 4;
    bool attachment = 5;
}

message SubmissionResult {
    int32 finalScore = 1;
    string error = 2;
    string finalOutput = 3;
}